src/ogdc_runner/recipe/viz_workflow.py

```diff
@@ -0,0 +1,319 @@
+            name="batch-input",
+            path="/mnt/workflow/input/ice_basins.gpkg",
+            url="{{inputs.parameters.input_url}}",
+        ),
+    ],
+    outputs=[
+        Artifact(
+            name="batch-output",
+            path="/mnt/workflow/output/batch",
+        ),
+    ],
+    image="ghcr.io/rushirajnenuji/viz-staging:latest",
+    command=["python"],
+    volume_mounts=[
+        VolumeMount(name="qgnet-ogdc-workflow-pvc", mount_path="/mnt/workflow")
+    ],
+)
+def batch_process(input_url, num_features) -> None:  # type: ignore[no-untyped-def] # noqa: ARG001
+    """Processes data in batches."""
+    import json
+    import sys
+    from pathlib import Path
+
+    import geopandas as gpd  # type: ignore[import-not-found]
+
+    # Redirect print statements to stderr instead of stdout
+    # This way they won't interfere with the JSON output
+    def print_log(message: str) -> None:
+        print(message, file=sys.stderr)
+
+    gdf = gpd.read_file("{{inputs.artifacts.batch-input.path}}")
+    results = []
+    for idx, start in enumerate(range(0, len(gdf), num_features)):
+        output_fp = Path(
+            "{{outputs.artifacts.batch-output.path}}/" + f"chunk-{idx}.gpkg"
+        )
+        print_log(f"Writing chunk {idx} to {output_fp}")
+        output_fp.parent.mkdir(parents=True, exist_ok=True)
+        gdf[start : start + num_features].to_file(
+            filename=output_fp,
+            driver="GPKG",
+        )
+        results.append(str(output_fp))
+
+    # Output only the JSON to stdout
+    print(json.dumps(results))
+
+
+@script(
+    name="tiling",
+    inputs=[
+        Parameter(name="chunk-filepath"),
+    ],
+    image="ghcr.io/rushirajnenuji/viz-staging:latest",
+    command=["python"],
+    volume_mounts=[
+        VolumeMount(name="qgnet-ogdc-workflow-pvc", mount_path="/mnt/workflow")
+    ],
+)
+def tiling_process() -> None:
+    """Creates tiles from a geospatial data chunk."""
+    import json
+    import sys
+    from pathlib import Path
+
+    from pdgstaging import TileStager  # type: ignore[import-not-found]
+
+    # Log to stderr
+    def print_log(message: str) -> None:
+        print(message, file=sys.stderr)
+
+    # Read the viz-config.json from the PVC
+    workflow_config = json.loads(
+        Path("/mnt/workflow/config/viz-config.json").read_text()
+    )
+    stager = TileStager(workflow_config, check_footprints=False)
+    print_log("Staging chunk file")
+    print_log("{{inputs.parameters.chunk-filepath}}")
+    stager.stage("{{inputs.parameters.chunk-filepath}}")
+    print_log("Staging done")
+
+
+def make_and_submit_viz_workflow(
+    recipe_config: RecipeConfig,
+    wait: bool,
+    custom_image: str | None = None,
+    custom_tag: str | None = None,
+    custom_namespace: str | None = None,
+    update_global: bool = False,
+    enable_tiling: bool = False,
+    enable_rasterize: bool = False,
+    enable_3dtiles: bool = False,
+    num_features: int = 250,
```

Comment:
Again, I'm wondering if these extra options should be exposed at the recipe level. Do we want options like `enable_tiling` and `enable_3dtiles` to be exposed to users, or just for us as developers?